Initial repository setup

cameronwaterman opened the pull request:
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/systemlink-operations-handbook/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- First pass at README files
- Basic application structure
- Skeleton for usage documentation
- .tox file for automation
- Test infrastructure

### Why should this Pull Request be merged?

Initial repository setup requirements.

### What testing has been done?

- Ran tox - outcome as expected
- Links in .md files work

Commit:
Initial repo setup

- First pass at README files
- Basic application structure
- Skeleton for usage documentation
- Makefile for test, lint, build, deploy
- Test infrastructure

spanglerco commented on Makefile:
```diff
@@ -0,0 +1,27 @@
+pyfiles = slconf test *.py
```
Any particular reason to use make instead of something like tox?

cameronwaterman replied:
Most of the Python project structure guidelines and example projects I found were using make (Including Mark's, which I pulled this Makefile from). For what it's worth, I _really_ would like to avoid make. I'll take a look into tox, or some other alternative and respond back to this conversation afterwards,

spanglerco commented on docs/conf.py:
```diff
@@ -0,0 +1,69 @@
+# -- Project information -----------------------------------------------------
+
+project = 'slconf'
+copyright = f'20220-{datetime.date.today().year}, cameronwaterman< and contributors'
+author = 'cameronwaterman <>'
```
I believe NI owns the copyright. We use this setting in the Python client repo:

```suggestion
copyright = f'2020-{datetime.date.today().year}, National Instruments'
author = 'National Instruments'
```

cameronwaterman replied:
Well that makes sense. Is it still National Instruments, or just NI now? Not sure since this is a legal-related.

spanglerco commented on setup.py:
```diff
@@ -0,0 +1,46 @@
+    name='slconf',
+    packages=['slconf'],
+    version=__version__,
+    author='cameronwaterman',
+    author_email='cameron.waterman@ni.com',
```
Based on the Python clients:

```suggestion
    author="National Instruments",
    maintainer="Cameron Waterman",
    maintainer_email="cameron.waterman@ni.com",
```

spanglerco commented on setup.py:
```diff
@@ -0,0 +1,46 @@
+        'Topic :: Software Development :: Libraries :: Python Modules',
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+        'Topic :: Software Development :: Libraries :: Python Modules',
```
Duplicate

spanglerco commented on .gitignore:
```diff
@@ -0,0 +1,139 @@
+# Byte-compiled / optimized / DLL files
```
General comment, have we considered using https://github.com/ni/nisystemlink-clients-python as a starting point for all of this?

spanglerco commented:
Looks like the repo isn't configured to require PR approvals before merging yet :)

cameronwaterman commented:
> Looks like the repo isn't configured to require PR approvals before merging yet :)

Yes, I'll configure that. Related, I accidentally pushed the initial README.md and CONTRIBUTING.md directly to master. So please take a look at the existing content of those files as well.

Commit:
Address review feedback

Update to more closely match tooling and structure of other NI-owned Python repos

spanglerco commented on .flake8:
```diff
@@ -0,0 +1,17 @@
+    # D400 and D205 are because we sometimes split the first line of the docstring
+    *.py E203, W503, D100,D101,D102,D105,D106,D107, D400,D205, D415
+    # temporary, until code is complete and stuff like ObjectDisposedException is gone:
+    *.py F821
```
We can probably drop this unless we find a need for it.
```suggestion
```

spanglerco commented on setup.py:
```diff
@@ -1,46 +1,66 @@
+    script_dir = os.path.dirname(os.path.realpath(__file__))
+    script_dir = os.path.join(script_dir, name)
+    if not os.path.exists(os.path.join(script_dir, "VERSION")):
+        version = "0.1.1"
```
```suggestion
        version = "0.1.0"
```

spanglerco commented on setup.py:
```diff
@@ -1,46 +1,66 @@
-    install_requires=read_requirements('requirements.txt'),
-    tests_require=read_requirements('test-requirements.txt'),
+    install_requires=[],
+    tests_require=["pytest", "pytest-asyncio", "mypy"],
```
I believe the client repo only needed mypy as a tests require because it would invoke it for some of the tests.

```suggestion
    tests_require=["pytest", "pytest-asyncio"],
```

spanglerco reviewed (changes requested):
> What testing has been done?

Can you update your PR description, hopefully to say that you ran `tox` :)

Commit:
Minor updates from review feedback

spanglerco commented on tox.ini:
```diff
@@ -13,12 +13,10 @@ deps =
     py3: flake8
     py3: flake8-docstrings
     py3: flake8-import-order
-    mypy
```
Sorry for the misunderstanding, but we do want to run mypy from tox. My comment was that in the python client repo, there were pytests that also ran mypy during the tests, which I thought is why it shows up in tests_require given that tox has its own deps list.

cameronwaterman replied:
Ah I see, my bad. I thought we might want it but figured you had a reason for removing it. That makes much more sense. I'll update 😄

Commit:
Add mypy back for running from tox
